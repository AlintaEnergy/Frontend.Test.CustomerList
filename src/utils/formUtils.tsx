// eslint-disable-next-line no-useless-escape
export const nameRegex = /^[A-Za-z\s\'-]*$/;

// Accepting 04xx xxx xxx, (0x) xxxx xxxx, +61 xxx xxx xxx and without space
// eslint-disable-next-line no-useless-escape
export const phoneRegex = /^(?:\+?(61))? ?(?:\((?=.*\)))?(0?[2-57-8])\)? ?(\d\d(?:[- ](?=\d{3})|(?!\d\d[- ]?\d[- ]))\d\d[- ]?\d[- ]?\d{3})$/;

export const toHeaderCase = (input: string) => {
  const replaceInput = input.replace(/([A-Z])/g, " $1");
  const changedInput =
    replaceInput.charAt(0).toUpperCase() + replaceInput.slice(1);
  return changedInput;
};
