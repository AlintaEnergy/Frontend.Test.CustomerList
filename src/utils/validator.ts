export const validateEmptyField = (value:string, errorMessage: string) => {
    return !value ? errorMessage : undefined 
  }

export const validatePattern = (value: string, pattern: RegExp, errorMessage: string) => {
    return !pattern.test(value) ? errorMessage : undefined
  }