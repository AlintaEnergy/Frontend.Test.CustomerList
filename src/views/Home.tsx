import * as React from "react";
import { useSelector, shallowEqual, useDispatch } from "react-redux";
import { Customer } from "../components/Customer/Customer";
import { AddCustomerForm } from "../components/AddCustomer/AddCustomerForm";
import { Dispatch } from "redux";
import { CustomerState, ICustomer } from "../types/types";
import { addCustomer, removeCustomer } from "../redux/actions/customerActions";

const Home: React.FC = () => {
  const customers: readonly ICustomer[] = useSelector(
    (state: CustomerState) => state.customers,
    shallowEqual
  );

  const dispatch: Dispatch<any> = useDispatch();

  const saveCustomer = React.useCallback(
    (customer: ICustomer) => dispatch(addCustomer(customer)),
    [dispatch]
  );

  return (
    <>
      <AddCustomerForm saveCustomer={saveCustomer} />
      <Customer customers={customers} removeCustomer={removeCustomer} />
    </>
  );
};

export default Home;
