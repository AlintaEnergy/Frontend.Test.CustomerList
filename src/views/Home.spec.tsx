import * as React from "react";
import { render } from "../utils/testUtils";
import  Home  from "./Home";

describe("<Home />", () => {
  test("should render a <Home />", () => {
    const wrapper = render(<Home />);
    expect(wrapper.container).toMatchSnapshot();
  });
});