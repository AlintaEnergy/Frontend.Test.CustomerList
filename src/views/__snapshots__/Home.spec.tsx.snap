// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render a <Home /> 1`] = `
<div>
  <form
    action="#"
    class="sc-gKAaRy byIIAQ"
  >
    <label
      class="sc-iCoGMd kTOdHT"
      for="firstName"
    >
      First Name *
    </label>
    <input
      class="sc-fujyAs byvWZq"
      id="firstName"
      name="firstName"
      placeholder="John"
      value=""
    />
    <label
      class="sc-iCoGMd kTOdHT"
      for="lastName"
    >
      Last Name *
    </label>
    <input
      class="sc-fujyAs byvWZq"
      id="lastName"
      name="lastName"
      placeholder="Doe"
      value=""
    />
    <label
      class="sc-iCoGMd kTOdHT"
      for="phoneNumber"
    >
      Phone Number *
    </label>
    <input
      class="sc-fujyAs byvWZq"
      id="phoneNumber"
      name="phoneNumber"
      placeholder="0400000000"
      type="tel"
      value=""
    />
    <button
      class="sc-pNWdM jwsycl"
      type="submit"
    >
      Add Customer
    </button>
  </form>
  <div
    class="sc-bdnxRM dqByRi"
  >
    <table
      class="sc-gtsrHT chztWe"
    >
      <thead>
        <tr
          class="sc-hKFxyN kUEvVU"
        >
          <th
            class="sc-dlnjwi iBDsHx"
          >
            Id
          </th>
          <th
            class="sc-dlnjwi iBDsHx"
          >
            First Name
          </th>
          <th
            class="sc-dlnjwi iBDsHx"
          >
            Last Name
          </th>
          <th
            class="sc-dlnjwi iBDsHx"
          >
            Phone Number
          </th>
          <th
            class="sc-dlnjwi iBDsHx"
          >
            Action
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="sc-hKFxyN kUEvVU"
        >
          <td
            class="sc-eCApnc bgLJTS"
          >
            1
          </td>
          <td
            class="sc-eCApnc bgLJTS"
          >
            Charles
          </td>
          <td
            class="sc-eCApnc bgLJTS"
          >
            Babbage
          </td>
          <td
            class="sc-eCApnc bgLJTS"
          >
            0412 123 123
          </td>
          <td
            class="sc-eCApnc bgLJTS"
          >
            <button
              class="sc-jSFjdj glHaqL"
            >
              Delete
            </button>
          </td>
        </tr>
        <tr
          class="sc-hKFxyN kUEvVU"
        >
          <td
            class="sc-eCApnc bgLJTS"
          >
            2
          </td>
          <td
            class="sc-eCApnc bgLJTS"
          >
            Alan
          </td>
          <td
            class="sc-eCApnc bgLJTS"
          >
            Turing
          </td>
          <td
            class="sc-eCApnc bgLJTS"
          >
            (03) 9599 1234
          </td>
          <td
            class="sc-eCApnc bgLJTS"
          >
            <button
              class="sc-jSFjdj glHaqL"
            >
              Delete
            </button>
          </td>
        </tr>
        <tr
          class="sc-hKFxyN kUEvVU"
        >
          <td
            class="sc-eCApnc bgLJTS"
          >
            3
          </td>
          <td
            class="sc-eCApnc bgLJTS"
          >
            Ada
          </td>
          <td
            class="sc-eCApnc bgLJTS"
          >
            Lovelace
          </td>
          <td
            class="sc-eCApnc bgLJTS"
          >
            +61 423 345 567
          </td>
          <td
            class="sc-eCApnc bgLJTS"
          >
            <button
              class="sc-jSFjdj glHaqL"
            >
              Delete
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
